megamek/src/megamek/server/totalwarfare/TWGameManager.java

```diff
@@ -10605,6 +10605,42 @@ void resolveAllButWeaponAttacks() {
         }
     }
 
+    void resolveBoobyTraps() {
+        Vector<Report> vDesc = new Vector<>();
+        Report r;
+        for (Entity e : game.getEntitiesVector()) {
+            if (e.hasBoobyTrap() &&
+                      e.getBoobyTrap().curMode().isArmed() &&
+                      e.isBoobyTrapInitiated()) {
+                int damage = e.getBoobyTrapDamage();
+                r = new Report(11000, Report.PUBLIC);
+                r.subject = e.getId();
+                r.addDesc(e);
+                r.indent(2);
+                r.add(damage);
+                vDesc.add(r);
+                if (e instanceof Mek mek) {
+                    if (mek.isAutoEject() &&
```

Comment:
If you don't enable autoeject, does that really mean you want your pilot to explode with the booby trap?

Reply:
Incomplete PR, it should eject, and I think there is one modifier missing.